addon/addon/models/sprite-snapshot-node-builder.ts

```diff
@@ -307,16 +307,53 @@ export class SpriteSnapshotNodeBuilder {
 
     // Classify non-natural KeptSprites
     for (let insertedSpriteModifier of classifiedInsertedSpriteModifiers) {
+      let counterpartSpriteModifier: SpriteModifier | undefined;
+
+      // find a suitable RemovedSprite counterpart if any
+      let removedSpriteModifiers = [...classifiedRemovedSpriteModifiers].filter(
+        (removedSpriteModifier) =>
+          new SpriteIdentifier(
+            insertedSpriteModifier.id,
+            insertedSpriteModifier.role
+          ).equals(
+            new SpriteIdentifier(
+              removedSpriteModifier.id,
+              removedSpriteModifier.role
+            )
+          )
+      );
+
+      assert(
+        'Multiple matching removedSpriteModifiers found',
+        removedSpriteModifiers.length < 2
+      );
+
+      let removedSpriteModifier = removedSpriteModifiers[0] as SpriteModifier;
+      if (removedSpriteModifier) {
+        classifiedRemovedSpriteModifiers.delete(removedSpriteModifier);
+      }
+
       let intermediateSprite = intermediateSprites.get(
         new SpriteIdentifier(
           insertedSpriteModifier.id,
           insertedSpriteModifier.role
         ).toString()
       );
-      if (intermediateSprite) {
+
+      if (intermediateSprite || removedSpriteModifier) {
+        classifiedKeptSpriteModifiers.add(insertedSpriteModifier);
+        classifiedInsertedSpriteModifiers.delete(insertedSpriteModifier);
+
+        // a matching IntermediateSprite always wins from a RemovedSprite counterpart
```

Comment:
The reason for this is that the intermediate sprite is more representative of the current (mid-animation) state of the element in the UI, should we mention this in this comment?